File: solutions/javascript/bob/3/bob.js
export const hey = (message) => {
  const trimmedMessage = message.trim();

  if (/^\s*$/.test(trimmedMessage)) { //^ start of string, \s* any number of whitespace characters, $ end of string
    return "Fine. Be that way!";
  }

  const isQuestion = /\?$/.test(trimmedMessage); //? end of string
  const isYelling = /^[^a-z]*[A-Z][^a-z]*$/.test (trimmedMessage); //^ start of string, [^a-z]* any number of non-lowercase characters, [A-Z] one uppercase character, [^a-z]* any number of non-lowercase characters, $ end of string

/*   return isQuestion
    ? isYelling
      ? "Calm down, I know what I'm doing!"
      : "Sure."
    : isYelling
    ? "Whoa, chill out!"
    : "Whatever."; */

    switch (true) {
      case isQuestion && isYelling:
        return "Calm down, I know what I'm doing!";
      case isQuestion:
        return "Sure.";
      case isYelling:
        return "Whoa, chill out!";
      default:
        return "Whatever.";
    }
};

File: solutions/csharp/bob/3/Bob.cs
using System;
public static class Bob
{
    private static bool IsEmpty(string input)
    {
        return string.IsNullOrWhiteSpace(input);
    }

    private static bool IsYellQuestion(string input)
    {
        return input.EndsWith('?') && (input.ToUpper() == input && input.ToLower() != input);
    }

    private static bool IsQuestion(string input)
    {
        return input.EndsWith('?');
    }

    private static bool IsYell(string input)
    {
        return input.ToUpper() == input && input.ToLower() != input;
    }

    public static string Response(string statement)
    {


        string trimString = statement.Trim();


        if (IsEmpty(trimString))
        {
            return "Fine. Be that way!";
        }

        if (IsYellQuestion(trimString))
        {
            return "Calm down, I know what I'm doing!";
        }
        if (IsQuestion(trimString))
        {
            return "Sure.";
        }
        if (IsYell(trimString))
        {
            return "Whoa, chill out!";
        }


        return "Whatever.";

    }


}